### PR TITLE
fix: refactor escapeHtml fn to cast nullish values to empty string

### PR DESCRIPTION
**Pull request by mostafa-raafat**

### CODE REVIEW

**Comment by Matwog** [comment]
@mostafa-raafat Could you also please add a fallback value.

```
  function applyMarkdown(translation = '') {
    // Escape underscores.
    // (Since we use underscores to denote interpolations, we have to
    // exclude them from the markdown notation. Use asterisk (*) instead.)
    let finalTranslation = marked(translation.replace(/_/g, '\\_'))

    // remove single, outer wrapping <p>-tag
    if (isWrappedInPTag(finalTranslation)) {
      // last occurrence of <p> is at the start, first occurrence of </p> is a the very end
      finalTranslation = finalTranslation.substring(
        3,
        finalTranslation.length - 5
      )
    }

    return finalTranslation.replace(/\\_/g, '_')
  }
```

File: src/translate.js [hunk 1]
```diff
@@ -70,7 +70,7 @@ export default (singleton) => {
     )
   }
 
-  function applyMarkdown(translation) {
+  function applyMarkdown(translation = '') {
```
Comment (Matwog) [inline src/translate.js:73]: **hint:** adds fallback

File: src/translate.js [hunk 2]
```diff
@@ -93,7 +93,7 @@ export default (singleton) => {
     return <span key={key} dangerouslySetInnerHTML={{ __html: html }} />
   }
 
-  function insertInterpolations(translation, options) {
+  function insertInterpolations(translation = '', options) {
```
Comment (Matwog) [inline src/translate.js:96]: **hint:** adds fallback

File: src/translate.js [hunk 3]
```diff
@@ -171,16 +171,26 @@ export default (singleton) => {
...
+  }
+
+  // Used to match HTML entities and HTML characters.
+  const reUnescapedHtml = /[&<>"']/g
+  // Cast (null,undefined,[] and 0 to empty string => '')
+  const reHasUnescapedHtml = RegExp(reUnescapedHtml.source)
+
+  return unsafe && reHasUnescapedHtml.test(unsafe)
+    ? unsafe.replace(reUnescapedHtml, (chr) => htmlEscapes[chr])
+    : unsafe || ''
```
Comment (Matwog) [inline src/translate.js:193]: **hint:** adds fallback

**Review by atilafassina — APPROVED** [review]
🏆

File: test/specs/i18n.spec.js [hunk 4]
```diff
@@ -215,7 +212,21 @@ describe('i18n', () => {
...
+    it('should handle undefined values with markdown', () => {
+      expect(
+        i18n(undefined, {
+          markdown: true,
+        })
+      ).toStrictEqual([])
+    })
+
+    it('should handle undefined values without markdown', () => {
+      expect(i18n(undefined, {})).toStrictEqual([])
```
Comment (Matwog) [inline test/specs/i18n.spec.js:229]: **hint:** adds tests

**Pull request merged**